### PR TITLE
fixes log levels, adds a few tests

### PR DESCRIPTION
**Pull request by wgahnagl**
Signed-off-by: Skyler Clark <wgahnagl@protonmail.com>

```release-note
none
```

### CODE REVIEW

**Comment by haircommander** [comment]
@saschagrunert now the release note plugin is enabled on the org, we get hit here too. do you know if it's possible to only require it in cri-o repo?

**Comment by saschagrunert** [comment]
> @saschagrunert now the release note plugin is enabled on the org, we get hit here too. do you know if it's possible to only require it in cri-o repo?

Yeah it should be possible, OTOH we could also move the PR template over here, WDYT?

**Comment by wgahnagl** [comment]
@umohnani8 @saschagrunert @saschagrunert @mrunalp  PTAL

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by umohnani8** [comment]
LGTM

**Comment by haircommander** [comment]
LGTM

File: pkg/ocicni/ocicni.go [hunk 1]
```diff
@@ -105,7 +105,7 @@ func (plugin *cniNetworkPlugin) podUnlock(podNetwork PodNetwork) {
 	fullPodName := buildFullPodName(podNetwork)
 	lock, ok := plugin.pods[fullPodName]
 	if !ok {
-		logrus.Warningf("Unbalanced pod lock unref for %s", fullPodName)
+		logrus.Warningf("Cannot find reference in refcount map for %s. Refcount cannot be determined.", fullPodName)
```
Comment (fidencio) [inline pkg/ocicni/ocicni.go:108]: This looks like something fatal. Wouldn't be better to have it as Errorf if that's really the case?

File: pkg/ocicni/ocicni.go [hunk 2]
```diff
@@ -278,13 +278,13 @@ func loadNetworks(confDir string, cni *libcni.CNIConfig) (map[string]*cniNetwork
 		if strings.HasSuffix(confFile, ".conflist") {
 			confList, err = libcni.ConfListFromFile(confFile)
 			if err != nil {
-				logrus.Warningf("Error loading CNI config list file %s: %v", confFile, err)
+				logrus.Errorf("Error loading CNI config list file %s: %v", confFile, err)
```
Comment (fidencio) [inline pkg/ocicni/ocicni.go:281]: I'm slightly new to this code base, so I'd like to understand what's the advantage of switching the "Warning" to "Error", considering this is not a fatal issue.

File: pkg/ocicni/ocicni.go [hunk 3]
```diff
@@ -278,13 +278,13 @@ func loadNetworks(confDir string, cni *libcni.CNIConfig) (map[string]*cniNetwork
...
 				continue
 			}
 		} else {
 			conf, err := libcni.ConfFromFile(confFile)
 			if err != nil {
-				logrus.Warningf("Error loading CNI config file %s: %v", confFile, err)
+				logrus.Errorf("Error loading CNI config file %s: %v", confFile, err)
```
Comment (fidencio) [inline pkg/ocicni/ocicni.go:287]: Same question here about the change from "Warning" to "Error".

File: pkg/ocicni/ocicni.go [hunk 4]
```diff
@@ -293,7 +293,7 @@ func loadNetworks(confDir string, cni *libcni.CNIConfig) (map[string]*cniNetwork
 			}
 			confList, err = libcni.ConfListFromConf(conf)
 			if err != nil {
-				logrus.Warningf("Error converting CNI config file %s to list: %v", confFile, err)
+				logrus.Errorf("Error converting CNI config file %s to list: %v", confFile, err)
```
Comment (fidencio) [inline pkg/ocicni/ocicni.go:296]: Same question here about the change from "Warning" to "Error".

File: pkg/ocicni/ocicni.go [hunk 5]
```diff
@@ -616,7 +616,7 @@ func (plugin *cniNetworkPlugin) getCachedNetworkInfo(containerID string) ([]NetA
 		cacheFile := filepath.Join(dirPath, fname)
 		bytes, err := ioutil.ReadFile(cacheFile)
 		if err != nil {
-			logrus.Warningf("failed to read CNI cache file %s: %v", cacheFile, err)
+			logrus.Errorf("failed to read CNI cache file %s: %v", cacheFile, err)
```
Comment (fidencio) [inline pkg/ocicni/ocicni.go:619]: Same question here about the change from "Warning" to "Error".

File: pkg/ocicni/ocicni.go [hunk 6]
```diff
@@ -628,7 +628,7 @@ func (plugin *cniNetworkPlugin) getCachedNetworkInfo(containerID string) ([]NetA
 		}{}
 
 		if err := json.Unmarshal(bytes, &cachedInfo); err != nil {
-			logrus.Warningf("failed to unmarshal CNI cache file %s: %v", cacheFile, err)
+			logrus.Errorf("failed to unmarshal CNI cache file %s: %v", cacheFile, err)
```
Comment (fidencio) [inline pkg/ocicni/ocicni.go:631]: Same question here about the change from "Warning" to "Error".

**Review by fidencio — COMMENTED** [review]
@wgahnagl, I've added a few comments. Please, bear with me as I'm slightly new to the code base, so the questions may not actually make sense.

File: pkg/ocicni/ocicni.go (hunk 1 above)
Comment (wgahnagl) [reply pkg/ocicni/ocicni.go]: fixed!

File: pkg/ocicni/ocicni.go (hunk 2 above)
Comment (wgahnagl) [reply pkg/ocicni/ocicni.go:281]: fixed!

File: pkg/ocicni/ocicni.go (hunk 3 above)
Comment (wgahnagl) [reply pkg/ocicni/ocicni.go:287]: fixed!

File: pkg/ocicni/ocicni.go (hunk 5 above)
Comment (wgahnagl) [reply pkg/ocicni/ocicni.go:619]: fixed!

File: pkg/ocicni/ocicni.go (hunk 6 above)
Comment (wgahnagl) [reply pkg/ocicni/ocicni.go:631]: fixed!

**Comment by umohnani8** [comment]
@wgahnagl needs a rebase

**Comment by umohnani8** [comment]
@haircommander PTAL

**Comment by haircommander** [comment]
I think you'll need to rebase on master, as it looks like there'll be some conflic t with https://github.com/cri-o/ocicni/pull/82

**Comment by wgahnagl** [comment]
II've rebased, but there's no updates left to do!

**Comment by umohnani8** [comment]
@haircommander github is not complaining about any conflicts as well, so this should be ready to go :)

**Comment by haircommander** [comment]
LGTM, @rhatdan PTAL since you just worked in this area

**Comment by rhatdan** [comment]
/lgtm
This is fine, although I would prefer if errors were returned rather then just logged.

File: pkg/ocicni/ocicni.go (hunk 2 above)
Comment (fidencio) [reply pkg/ocicni/ocicni.go:278]: Hey @wgahnagl , you mentioned it was fixed but it seems that it's not.  The diff still shows the change switching from "Warning"  to "Error".

File: pkg/ocicni/ocicni.go (hunk 3 above)
Comment (fidencio) [reply pkg/ocicni/ocicni.go:284]: Same here.

File: pkg/ocicni/ocicni.go (hunk 4 above)
Comment (fidencio) [reply pkg/ocicni/ocicni.go:293]: Same here.

File: pkg/ocicni/ocicni.go (hunk 5 above)
Comment (fidencio) [reply pkg/ocicni/ocicni.go:573]: Same here,

File: pkg/ocicni/ocicni.go (hunk 6 above)
Comment (fidencio) [reply pkg/ocicni/ocicni.go:585]: And same here.

**Review by fidencio — CHANGES_REQUESTED** [review]
@wgahnagl, let me take my LGTM back (sorry about that).

I think you did the changes locally but ended up not pishing them.

**Pull request merged**